crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -699,12 +764,15 @@ fn build_lattice_methods_by_wit_interface(
             // across the lattice, in a <CamelCaseModule><CamelCaseInterface><CamelCaseFunctionName> pattern
             // (ex. MessagingConsumerRequestMultiInvocation)
             let lattice_method_name = LitStr::new(
-                format!(
-                    "{}{}/Message.{}",
-                    wit_ns,
-                    wit_package,
-                    f.sig.ident.to_string().to_upper_camel_case()
-                )
+                match (wit_ns, wit_package) {
+                    ("wasmcloud", "keyvalue") => {
+                        format!("KeyValue.{}", f.sig.ident.to_string().to_upper_camel_case())
```

Comment:
I was hoping we could avoid hardcoding here and turn `key-value` interface name into `KeyValue`

Reply:
Yup, that isn't hard -- but wasn't necessary for the experimentation I was doing. 

This shouldn't be changed here but needs to be fed through as an option to provider bindgen itself to configure how these names are generated, since there are multiple ways (one for legacy providers and another for new providers -- the `<ns>:<pkg>/<interface>.<fn>` syntax). 

If that extra option gets added in this PR, then this code will be changed to use the new logic